plugins/editqml/qml/PaletteContainer.qml

```diff
@@ -26,22 +26,29 @@ Rectangle{
     property bool compact: true
     property bool isBuilder : false
 
-    property Item child : null
+    property Item child : palette ? palette.item : null
     property QtObject palette : null
 
     property bool isCompactVertical: child && child.height > 48
 
     color: "black"
 
-    property string name : ''
-    property string type : ''
+    property string name : palette ? palette.name : ''
+    property string type : palette ? palette.type : ''
     property string title : type + ' - ' + name
     property var cursorRectangle : null
     property var editorPosition : null
     property var editor: null
     property Item pane: null
     property Item dragPane: null
 
+    onEditingFragmentChanged: {
+        if (!editingFragment) return

```

Comment:
### Problem:

```
Item{}
Item{}
```

And we shape the 2nd `item`, and then we start modifying the 1st item.

```js
Item#1{
    width: 100
}
Item#2{}
```

Now, `item#2` has now changed its position and cursor rectangle.

The position is held in place by the EditFragment. The `EditFragment` will always have the modified/latest `position`, so it should be captured from there.

Check where `editorPosition` is used, and use the `editFragment.position` instead. 

And, get the `cursorRectangle` if need, from `editor.textEdit.positionToRectangle(editFragment.position)`